llvm/test/lab2/prokofev_kirill/mytest.ll

```diff
@@ -0,0 +1,164 @@
+; RUN: split-file %s %t
+; RUN: opt -passes="prokofev-replace-mul-with-shift" -S %t/test1.ll | FileCheck %t/test1.ll
```

Comment:
Please add an option `-mul-shift-const-only` which if set to true limits replacement of multiplication with shl only in case if pow_of_two operand of multiplication is a constant.

You can look into llvm/examples/Bye/Bye.cpp regarding custom options adding.

Reply:
Added